tests/src/test_set.cpp

```diff
@@ -85,6 +99,19 @@ SCENARIO( "a Set can be interacted with by consuming classes even when const", "
 				CHECK( secondPtr == firstPtr );
 			}
 		}
+
+		WHEN( "find(key) is called" ) {
+			const auto cIt = s.find(key);
+			THEN( "the Set does not increase in size" ) {
+				CHECK( s.size() == 1 );
+			}
+			THEN( "the iterator does point to the begin position" ) {
+				CHECK( cIt == s.begin() );
+			}			
```

Comment:
The test should be that the found value is correct. Not that it is the same as `begin()`.

(Consider if we had two elements, and we called find with the second element's key. This test title says we should have gotten the first element, not the one we were looking for.)